src/native/fs/fs_napi.cpp

```diff
@@ -654,9 +661,9 @@ struct Stat : public FSWorker
         : FSWorker(info)
     {
         _path = info[1].As<Napi::String>();
-        if ((int)info.Length() == 3) {
+        if (info[2].ToBoolean()) {
```

Comment:
We had the check for number of arguments passed earlier but here we are missing it. I might be wrong but I think here if the function is invoked with incorrect number of arguments then that might throw error or worse (don't know if runtime bound checks happens when node gyp compiles) we might read invalid memory?

👆 assumes that we want to fail nicely if the function is invoked with incorrect number of arguments.

Reply:
Right, but the point is that it acts just like an array in javascript, when you access an index above length, it just returns undefined. So there is no real need to check for length specifically, but just that it evaluates to a non-falsy value.

Reply:
Ah, that's neat.